packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketInHistoryList.tsx

```diff
@@ -162,18 +163,28 @@ export interface RedPacketInHistoryListProps {
 export function RedPacketInHistoryList(props: RedPacketInHistoryListProps) {
     const { history, onSelect } = props
     const t = useI18N()
-    const { classes, cx } = useStyles()
+    const [isVisible, setVisible] = useState(false)
+
+    const ref = useRef<HTMLLIElement | null>(null)
+    const entry = useIntersectionObserver(ref, {})
     const { account, chainId } = useChainContext<NetworkPluginID.PLUGIN_EVM>()
     const isSmall = useMediaQuery((theme: Theme) => theme.breakpoints.down('sm'))
-    const { value: receipt } = useCreateRedPacketReceipt(history.txid)
+    const { value: receipt } = useCreateRedPacketReceipt(isVisible ? history.txid : '')
 
     const rpid = receipt?.rpid ?? ''
     const creation_time = receipt?.creation_time ?? 0
 
+    const { classes, cx } = useStyles({ isVisible: isVisible && Boolean(rpid) })
+
     const patchedHistory: RedPacketJSONPayload | RedPacketJSONPayloadFromChain = useMemo(
         () => ({ ...props.history, rpid, creation_time }),
         [props.history, rpid, creation_time],
     )
+
+    useEffect(() => {
+        if (entry?.isIntersecting) setVisible(true)
```

Comment:
The `setVisible` is used only once. Why not compute `visible` from `entry` directly?

Reply:
I think rename `visible` to `visibled` is less misunderstanding.